src/Form/FormAnalyzer.js

```diff
@@ -3,6 +3,13 @@ import { constants } from '../constants.js'
 import { matchingConfiguration } from './matching-configuration.js'
 import { getText, isLikelyASubmitButton } from '../autofill-utils.js'
 
+const negativeRegex = new RegExp(/sign(ing)?.?in(?!g)|log.?in|unsubscri|(forgot(ten)?|reset) (your )?password|password forgotten/i)
+const positiveRegex = new RegExp(
+    /sign(ing)?.?up|join|\bregist(er|ration)|newsletter|\bsubscri(be|ption)|contact|create|start|settings|preferences|profile|update|checkout|guest|purchase|buy|order|schedule|estimate|request|new.?customer|(confirm|retype|repeat) password/i
+)
+const conservativePositiveRegex = new RegExp(/sign.?up|join|register|newsletter|subscri(be|ption)|settings|preferences|profile|update/i)
+const strictPositiveRegex = new RegExp(/sign.?up|join|register|settings|preferences|profile|update/i)
+
```

Comment:
I've moved these regexes up here so they're not re-created every time we run `updateSignal` which can be several times on some pages.